src/components/Media/Media.tsx

```diff
@@ -1,35 +1,88 @@
+import {
+  CanvasNormalized,
+  AnnotationPageNormalized,
+  AnnotationNormalized,
+  ContentResource,
+} from "@hyperion-framework/types";
 
-interface Props {
-    items: object;
-    activeItem: number;
+interface MediaProps {
+  items: object[];
+  activeItem: number;
 }
 
-const Media: React.FC<Props> = ({ items, activeItem }) => {
-    return (
-        <MediaWrapper>
-            <MediaItem index={0} label="" thumbnailId="" active={false} />
-            <MediaItem index={1} label="" thumbnailId="" active={true} />
-            <MediaItem index={2} label="" thumbnailId="" active={false} />
-            <MediaItem index={3} label="" thumbnailId="" active={false} />
-            <MediaItem index={4} label="" thumbnailId="" active={false} />
-            <MediaItem index={5} label="" thumbnailId="" active={false} />
-            <MediaItem index={6} label="" thumbnailId="" active={false} />
-            <MediaItem index={7} label="" thumbnailId="" active={false} />
-            <MediaItem index={8} label="" thumbnailId="" active={false} />
-        </MediaWrapper>
-    );
+const Media: React.FC<MediaProps> = ({ items, activeItem }) => {
+  const dispatch: any = useViewerDispatch();
+  const state: any = useViewerState();
+  const { activeCanvas, vault } = state;
+
+  const annotationMotivation = "painting";
+  const contentResourceType = ["Sound", "Video"];
+
+  console.log(activeCanvas);
+
+  const onChange = (canvasId: string) => {
+    if (activeCanvas !== canvasId)
+      dispatch({
+        type: "updateActiveCanvas",
+        canvasId: canvasId,
+      });
+  };
+
+  return (
+    <MediaWrapper>
+      {items.map((item: object, key: number) => {
+        // 1. get canvas
+        const canvas: CanvasNormalized = vault.fromRef(item);
+        // 2. get annotationPage
+        const annotationPage: AnnotationPageNormalized = vault.fromRef(
+          canvas.items[0],
+        );
+        // 3.get annotations
+        const annotations: Array<AnnotationNormalized> = vault.allFromRef(
+          annotationPage.items,
+        );
+
+        // 4.determine if W3C annotation is motivation painting
+        for (const annotation of annotations) {
+          if (annotation.motivation.includes(annotationMotivation)) {
+            // 5.get contentResource (the annotation body)
+            const contentResource: ContentResource = vault.fromRef(
+              annotation.body[0],
+            );
+            // 6. render media item if annotation target matches original
+            //    canvas ID and contentResource type is Sound or Video
+            if (
+              annotation.target === item.id &&
+              contentResourceType.includes(contentResource.type)
+            )
+              return (
+                <MediaItem
+                  key={key}
+                  canvas={canvas}
+                  active={true}
+                  onChange={(canvasId) => {
```

Comment:
I think this could just be `onChange={onChange}`?   Just semantics, but sometimes more clear to name our functions unique from React function names.  So maybe our `onChange()` becomes `handleChange()`?